pkg/controller/factory/factory.go

```diff
@@ -76,13 +83,26 @@ func (f *Factory) WithRuntimeObject() *Factory {
 	return f
 }
 
+// WithSyncContext allows to specify existing sync context to this factory.
+// This is useful during unit testing.
+func (f *Factory) WithSyncContext(ctx SyncContext) *Factory {
```

Comment:
can a factory exist without `SyncFunc`? Can a sync func exist without `SyncContext`?

Reply:
What is the default if there is any?

Reply:
SyncContext is set for each sync() call in case it contain the runtime object for example. Factory can exists without sync context if you don't call sync()...
This stub only exists so you can re-use existing event recorder (in case you have unit test with unit testing event recorder)...

If the sync context is not provided, factory create one for you.

Reply:
document that: `// ... If this is not called, a sync context is created by the factory`